src/util/AllocatorWithLimit.h

```diff
@@ -173,32 +177,37 @@ class AllocatorWithLimit {
     const auto bytesNeeded = n * sizeof(T);
     const bool wasEnoughLeft =
         memoryLeft_.ptr()->wlock()->decrease_if_enough_left_or_return_false(
-            bytesNeeded);
+            MemorySize::bytes(bytesNeeded));
     if (!wasEnoughLeft) {
       clearOnAllocation_(n);
-      memoryLeft_.ptr()->wlock()->decrease_if_enough_left_or_throw(bytesNeeded);
+      memoryLeft_.ptr()->wlock()->decrease_if_enough_left_or_throw(
+          MemorySize::bytes(bytesNeeded));
     }
     // the actual allocation
     return allocator_.allocate(n);
   }
+  // Overload for use with `MemorySize`.
+  T* allocate(MemorySize n) { return allocate(n.getBytes()); }
 
   // An allocator must have a function "deallocate" with exactly this signature.
   void deallocate(T* p, std::size_t n) {
     // free the memory
     allocator_.deallocate(p, n);
     // Update the amount of memory left.
-    memoryLeft_.ptr()->wlock()->increase(n * sizeof(T));
+    memoryLeft_.ptr()->wlock()->increase(MemorySize::bytes(n * sizeof(T)));
   }
+  // Overload for use with `MemorySize`.
+  void deallocate(T* p, MemorySize n) { deallocate(p, n.getBytes()); }
```

Comment:
Same here, can probably be removed (and should).

Reply:
See above.

Reply:
Deleted in [4e421e4](https://github.com/ad-freiburg/qlever/pull/1112/commits/4e421e48d7973e30d692cf282cade5eea8c808f5).